sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -74,10 +74,20 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   }
 
   /**
-   * @return All metrics containing metrics of this SparkPlan.
+   * @return Executor side metrics of this SparkPlan.
    */
   def metrics: Map[String, SQLMetric] = Map.empty
 
+  /**
+   * @return Driver side metrics of this SparkPlan which no need to send them to executor.
+   */
+  def driverMetrics: Map[String, SQLMetric] = Map.empty
+
+  /**
+   * @return All metrics of this SparkPlan.
+   */
+  def allMetrics: Map[String, SQLMetric] = metrics ++ driverMetrics
```

Comment:
it's only called once, can we inline it?

Reply:
No problem.